sympy/core/expr.py

```diff
@@ -4137,10 +4138,140 @@ def search_element(self, elem):
         return None
 
 
+def _power_gens(expr):
+    from sympy.ntheory.factor_ import perfect_power
+    # helper for generators to analyze the powers of the
+    # factors in expr
+    p = expr
+    if expr.is_Pow:
```

Comment:
- [ ] should any of this be moved to `decompose_power`...or that routine be used from here?